lib/rubocop/cop/rspec/capybara/specific_matcher.rb

```diff
@@ -56,7 +79,31 @@ def specific_matcher(arg)
           end
 
           def acceptable_pattern?(arg)
-            arg.match?(/\[.+=\w+\]/) || arg.match?(/[ >,+]/)
+            arg.match?(/[ >,+]/)
+          end
+
+          def specific_matcher_option?(arg, matcher)
+            # If `button[foo-bar_baz=foo][bar][baz]`:
+            # extract ["foo-bar_baz", "bar", "baz"]
+            attributes = arg.scan(/\[(.*?)\]/).flatten
+            return true if attributes.empty?
+
+            attributes.all? do |attr|
+              specific_matcher_options(matcher).include?(attr.split('=').first)
+            end
+          end
+
+          def specific_matcher_options(matcher)
+            case matcher
+            when 'button'
+              HAVE_BUTTON_OPTIONS
+            when 'link'
+              HAVE_LINK_OPTIONS
+            when 'table'
+              HAVE_TABLE_OPTIONS
+            when 'select'
+              HAVE_SELECT_OPTIONS
```

Comment:
should we default to [] if the matcher is not in the list (even though some other part might guarantee it's one of those)

Reply:
I added this because it is an unlikely handling, but it certainly seemed like it would be better to have.